src/components/uploads/queue/index.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * @author johnworth
+ *
+ * A table populated from tracked uploads. Used in the UploadQueue.
+ *
+ * @module uploads/queue
+ */
+
+import React from "react";
+
+import {
+    Avatar,
+    CircularProgress,
+    IconButton,
+    List,
+    ListItem,
+    ListItemAvatar,
+    ListItemSecondaryAction,
+    ListItemText,
+    useMediaQuery,
+} from "@material-ui/core";
+
+import {
+    CheckCircle as CheckCircleIcon,
+    Cancel as CancelIcon,
+    Error as ErrorIcon,
+    Description as DescriptionIcon,
+    Http as HttpIcon,
+} from "@material-ui/icons";
+
+import {
+    KindFile,
+    removeAction,
+    useUploadTrackingState,
+    useUploadTrackingDispatch,
+} from "../../../contexts/uploadTracking";
+
+import { useTheme, makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({
+    ellipsis: {
+        overflow: "hidden",
+        textOverflow: "ellipsis",
+        paddingRight: theme.spacing(1),
+    },
+}));
```

Comment:
does this ellipsis style work ? I only ask bcos I had to follow this https://stackoverflow.com/questions/17779293/css-text-overflow-ellipsis-not-working to get it working.

Reply:
Hmm, odd. It worked fine for me.